### PR TITLE
[Jenkins-master] Use Java 7 with custom ARGS.

### PR DESCRIPTION
**Pull request by benpatterson**

### CODE REVIEW

**Comment by benpatterson** [comment]
@jzoldak @edx/devops please take a look.

File: playbooks/roles/jenkins_master/tasks/main.yml [hunk 1]
```diff
@@ -38,6 +38,18 @@
 - name: stop Jenkins
   service: name=jenkins state=stopped
 
+- name: set jvm args
+  lineinfile: >
+    backup=yes
```
Comment (feanil) [inline playbooks/roles/jenkins_master/tasks/main.yml]: This should use the new style syntax like we do [here](https://github.com/edx/configuration/pull/2458/files#diff-d887fe1442a4be97ae2d3183ac49489eR28).

File: playbooks/roles/jenkins_master/meta/main.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ---
 dependencies:
   - common
-  - oraclejdk
+  - role: oraclejdk
+    tags: java
```
Comment (feanil) [inline playbooks/roles/jenkins_master/meta/main.yml:5]: Why tag here instead of in the oraclejdk role itself?

Comment (benpatterson) [inline playbooks/roles/jenkins_master/meta/main.yml:5]: Tagging here means I can run jenkins set-up, but only java stuff. So, in the case of dependencies, we don't run common. (Does that answer your question?)

File: playbooks/roles/jenkins_master/tasks/main.yml (hunk 1 above)
Comment (benpatterson) [inline playbooks/roles/jenkins_master/tasks/main.yml]: Done. Pls take another look.

**Comment by benpatterson** [comment]
@feanil ready for another pass

File: playbooks/roles/jenkins_master/meta/main.yml (hunk 2 above)
Comment (feanil) [inline playbooks/roles/jenkins_master/meta/main.yml:5]: If you tag in the oraclejdk role that should do the same, but this is a much smaller change to get you to do what you need so I think that's fine.

File: playbooks/roles/jenkins_master/tasks/main.yml [hunk 3]
```diff
@@ -38,6 +38,18 @@
 - name: stop Jenkins
   service: name=jenkins state=stopped
 
+- name: set jvm args
+  lineinfile:
+    backup: yes
+    dest: /etc/default/jenkins
+    regexp: '^JAVA_ARGS'
```
Comment (feanil) [inline playbooks/roles/jenkins_master/tasks/main.yml]: Is there only one var that starts with `JAVA_ARGS`?  Theres not other vars like `JAVA_ARGS_THREADING`?  I would add an `=` in the regex just to be safe.

**Comment by feanil** [comment]
One nit in the regex if you think it's worth fixing, but :+1: even if you don't want to fix it.

**Comment by jzoldak** [comment]
:+1:

**Comment by feanil** [comment]
:+1:

**Comment by benpatterson** [comment]
Fixed the regex, merging. Thanks guys.

**Pull request merged**